src/sparsezoo/analysis/helpers.py

```diff
@@ -12,40 +12,234 @@
-    "get_layer_param",
+    "get_node_weight",
+    "get_node_bias",
     "get_node_num_zeros_and_size",
     "get_zero_point",
     "is_four_block_sparse_layer",
     "is_parameterized_prunable_layer",
     "is_quantized_layer",
     "is_sparse_layer",
+    "get_num_dense_and_sparse_ops",
+    "NodeShape",
+    "extract_node_shapes",
 ]
 
 
-def get_initializer(model: ModelProto, initializer_name: str) -> numpy.ndarray:
+def get_node_attributes(node: NodeProto) -> Dict[str, Any]:
+    """
+    :param node: node to which the attributes belong to
+    :return: a dictionary of attribute name and value pairs
+    """
+    attributes = {}
+    for attribute in node.attribute:
+        attributes[attribute.name] = get_attribute_value(attribute)
+
+    return attributes
+
+
+def get_node_bias(model: ModelProto, node: NodeProto) -> numpy.ndarray:
+    """
+    Finds parameter value of node (the node weight)
+
+    :param model: model that contains the given node
+    :param node: node to which parameter belongs to
+    :return: a numpy array of param value, None if not found
+    """
+
+    def _get_node_bias_name(model: ModelProto, node: NodeProto) -> str:
+        if node.op_type in ["Conv", "Gemm"]:
+            return _get_node_input(node, 2, default=None)
+
+        if node.op_type == "QLinearConv":
+            return _get_node_input(node, 8, default=None)
+
+        return None
+
+    initializer_name = _get_node_bias_name(model, node)
+    if initializer_name is None:
+        return None
+
+    return get_initializer_value(model, node, initializer_name)
+
+
+def get_num_dense_and_sparse_ops(
+    model: ModelProto,
+    node: NodeProto,
+    node_shapes: Optional[Dict[str, NodeShape]] = None,
+    is_four_block_sparse: Optional[bool] = None,
+) -> Tuple[int, int]:
+    """
+    Gets an approximation of the number of floating point or integer operations
+
+    :param model: model that contains the given node
+    :param node: node which performs the operations
+    :param node_shapes: Optional dictionary of node shapes. If not supplied,
+        node_shapes will be computed
+    :param is_four_block_sparse: Optional boolean indicating if this node is four
+        block sparse. If not supplied, it be will be computed
+    :return: number of operations performed by node
+    """
+    if node_shapes is None:
+        node_shapes = extract_node_shapes(model)
+
+    node_shape = node_shapes.get(extract_node_id(node))
+    input_shapes = node_shape.input_shapes if node_shape is not None else None
+    output_shapes = node_shape.output_shapes if node_shape is not None else None
+
+    weight = get_node_weight(model, node)
+    bias = get_node_bias(model, node)
+    zero_point = get_zero_point(model, node)
+    if is_four_block_sparse is None:
+        is_four_block_sparse = is_four_block_sparse_layer(model, node)
+
+    node_attributes = get_node_attributes(node)
+
+    if node.op_type in ["Add", "Mul", "Div", "Sub", "Clip"] or node.op_type in [
```

Comment:
why do we split into two statements?
i.e.
why not `if node.op_type in list1 + list2` instead of `if node.op_type in list1 or list2`?